--- file_server.ts	2022-09-11 02:46:58.000000000 -0700
+++ mod.ts	2022-09-11 02:47:30.000000000 -0700
@@ -347,6 +347,8 @@
       }),
     );
   } else if (e instanceof Deno.errors.NotFound) {
+    if (globalThis.finalHandler) return globalThis.finalHandler(_req);
+
     return Promise.resolve(
       new Response(STATUS_TEXT[Status.NotFound], {
         status: Status.NotFound,
@@ -651,7 +653,7 @@
     : normalizedUrl;
 }
 
-function main() {
+export default function main() {
   const serverArgs = parse(Deno.args, {
     string: ["port", "host", "cert", "key"],
     boolean: ["help", "dir-listing", "dotfiles", "cors", "verbose"],
