--- file_server.ts	2022-09-10 13:37:52.000000000 -0700
+++ mod.ts	2022-09-10 13:42:51.000000000 -0700
@@ -347,6 +347,26 @@
       }),
     );
   } else if (e instanceof Deno.errors.NotFound) {
+    if (!_req.finalHandlerChecked) {
+      _req.finalHandlerChecked = true; // (avoids infinite recursion)
+      return (async () => {
+        if (!globalThis.finalHandler && globalThis.finalHandler !== false && Deno.args.includes('--handler')) {
+          try {
+            const handlerArg: string = Deno.args[1 + Deno.args.indexOf('--handler')];
+            console.warn(`using handler: ${Deno.cwd()}/${handlerArg}`)
+            globalThis.finalHandler = (await import(`${Deno.cwd()}/${handlerArg}`)).default
+          } catch (e) {
+            globalThis.finalHandler = false;
+          }
+        }
+        if (globalThis.finalHandler)
+          return globalThis.finalHandler(_req);
+
+        return serveFallback(_req, e); // re-call ourself, but this time we'll std 404 below
+      })();
+    }
+    delete _req.finalHandlerChecked;
+
     return Promise.resolve(
       new Response(STATUS_TEXT[Status.NotFound], {
         status: Status.NotFound,
