--- file_server.ts	2022-09-10 14:49:19.000000000 -0700
+++ mod.ts	2022-09-11 18:59:08.000000000 -0700
@@ -347,6 +347,8 @@
       }),
     );
   } else if (e instanceof Deno.errors.NotFound) {
+    if (globalThis.finalHandler) return globalThis.finalHandler(_req);
+
     return Promise.resolve(
       new Response(STATUS_TEXT[Status.NotFound], {
         status: Status.NotFound,
@@ -605,6 +607,7 @@
 
   if (opts.enableCors) {
     assert(response);
+    if (!response.headers.get('access-control-allow-origin'))
     response.headers.append("access-control-allow-origin", "*");
     response.headers.append(
       "access-control-allow-headers",
@@ -651,7 +654,7 @@
     : normalizedUrl;
 }
 
-function main() {
+export default function main() {
   const serverArgs = parse(Deno.args, {
     string: ["port", "host", "cert", "key"],
     boolean: ["help", "dir-listing", "dotfiles", "cors", "verbose"],
